### ISSUE TITLE
[Docs]: Remove old accessibility message from components

### ISSUE THREAD

**Issue by heloiselui** [carbon-design-system/carbon-website#4835]
### Project

https://www.carbondesignsystem.com

### Section

https://carbondesignsystem.com/components/progress-indicator/accessibility/

### Problem

The accessibility pages for Date Picker, Progress Indicator, and Structured List still show an old  message block saying that these components were tested against WCAG 2.1 and that high‑priority issues were found.
This message is outdated and no longer needed.

We can remove this:

```
<PageDescription>

The structured list React Carbon component has been tested against the latest
[W3C Web Content Accessibility Guidelines (WCAG) 2.1 Level A and AA success criteria](https://www.w3.org/TR/WCAG21/)
and violations have been identified as high priority issues. This document will
be updated when these accessibility issues are resolved.

</PageDescription>
```

> [!NOTE]  
> We need double check Date Picker, because we have some a11y issues open https://github.com/carbon-design-system/carbon/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22type%3A%20a11y%20%E2%99%BF%22%20date%20picker

### Solution

From: 
https://carbondesignsystem.com/components/date-picker/accessibility/
https://carbondesignsystem.com/components/progress-indicator/accessibility/
https://carbondesignsystem.com/components/structured-list/accessibility/

Code path: 
[date-picker/accessibility.mdx](https://github.com/carbon-design-system/carbon-website/blob/b18c6e478e7a008097159d4ccb73f142a9f51d0c/src/pages/components/date-picker/accessibility.mdx?plain=1#L22)

[progress-indicator/accessibility.mdx
](https://github.com/carbon-design-system/carbon-website/blob/b18c6e478e7a008097159d4ccb73f142a9f51d0c/src/pages/components/progress-indicator/accessibility.mdx?plain=1#L23)

[structured-list/accessibility.mdx](https://github.com/carbon-design-system/carbon-website/blob/b18c6e478e7a008097159d4ccb73f142a9f51d0c/src/pages/components/structured-list/accessibility.mdx?plain=1#L22C1-L22C18)

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
